pyatlan/client/workflow.py

```diff
@@ -0,0 +1,189 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 Atlan Pte. Ltd.
+from logging import Logger
+from time import sleep
+from typing import Optional, Protocol, Union, overload, runtime_checkable
+
+from pyatlan.client.constants import (
+    WORKFLOW_INDEX_RUN_SEARCH,
+    WORKFLOW_INDEX_SEARCH,
+    WORKFLOW_RUN,
+)
+from pyatlan.errors import ErrorCode
+from pyatlan.model.enums import AtlanWorkflowPhase, WorkflowPackage
+from pyatlan.model.search import Bool, NestedQuery, Prefix, Query, Term
+from pyatlan.model.workflow import (
+    ReRunRequest,
+    WorkflowResponse,
+    WorkflowRunResponse,
+    WorkflowSearchRequest,
+    WorkflowSearchResponse,
+    WorkflowSearchResult,
+    WorkflowSearchResultDetail,
+)
+
+MONITOR_SLEEP_SECONDS = 5
+
+
+@runtime_checkable
+class ApiCaller(Protocol):
+    def _call_api(
+        self, api, query_params=None, request_obj=None, exclude_unset: bool = True
+    ):
+        pass
+
+
+class WorkflowClient:
+    """
+    This class can be used to retrieve information and re_run workflows. This class does not need to be instantiated
+    directly by can be obtained through the workflow property of AtlanClient.
+    """
+
+    def __init__(self, client: ApiCaller):
+        if not isinstance(client, ApiCaller):
+            raise ErrorCode.INVALID_PARAMETER_TYPE.exception_with_parameters(
+                "client", "ApiCaller"
+            )
+        self._client = client
+
+    def find_by_type(
+        self, prefix: WorkflowPackage, max_results: int = 10
+    ) -> list[WorkflowSearchResult]:
+        """
+        Find workflows based on their type (prefix). Note: Only workflows that have been run will be found.
+
+        :param prefix: name of the specific workflow re_run to find (for example ATLAN_CONNECTION_DELETE)
```

Comment:
name of the specific **type of** workflow ~~re_run~~ to find (for example ATLAN_CONNECTION_DELETE)